code/modules/projectiles/gun_system.dm

```diff
@@ -1534,6 +1574,12 @@
 	if(CHECK_BITFIELD(flags_gun_features, GUN_IS_ATTACHMENT) && !master_gun && CHECK_BITFIELD(flags_gun_features, GUN_ATTACHMENT_FIRE_ONLY))
 		to_chat(user, span_notice("You cannot fire [src] without it attached to a gun!"))
 		return FALSE
+	if(heat_level >= max_heat || overheated)
+		user.balloon_alert(user, "The gun is overheated!")
+		if(!overheat_timer)
+			overheated = TRUE
+			overheat_timer = addtimer(CALLBACK(src, .proc/cooldown), cooldown_time, TIMER_STOPPABLE)
```

Comment:
so you start the timer when you are trying to shoot ?

Reply:
yes

Reply:
That seems really frustrating